exporter/awsemfexporter/metric_declaration.go

```diff
@@ -186,8 +186,8 @@ func (lm *LabelMatcher) init() (err error) {
 	if len(lm.Separator) == 0 {
 		lm.Separator = ";"
 	}
-	lm.compiledRegex = regexp.MustCompile(lm.Regex)
-	return
+	lm.compiledRegex, err = regexp.Compile(lm.Regex)
```

Comment:
`MustCompile`  will panic if the [regexp is not compilable](https://pkg.go.dev/regexp#MustCompile). Therefore, using `Compile` which are the same with MustCompile but return readable error instead of panic